pkg/experiment/metastore/admin/admin.go

```diff
@@ -0,0 +1,258 @@
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/grafana/pyroscope/pkg/experiment/metastore/discovery"
+	"github.com/grafana/pyroscope/pkg/experiment/metastore/raftnode/raftnodepb"
+	httputil "github.com/grafana/pyroscope/pkg/util/http"
+)
+
+type formActionHandler func(http.ResponseWriter, *http.Request, string) error
+
+type MetastoreAdmin struct {
+	sharedClient raftnodepb.RaftNodeOpsServiceClient
```

Comment:
Please, consider a more specific name. It is the client that always directs requests to the cluster leader, call it `leaderClient`, or just `leader`. The type will tell the rest.